frontend/awx/common/PageFormMultiSelectAwxResource.tsx

```diff
@@ -65,11 +66,25 @@ export function PageFormMultiSelectAwxResource<
         return {
           remaining: response.count - response.results.length,
           options:
-            response.results?.map((resource) => ({
-              label: resource.name,
-              value: resource as PathValue<FormData, Name>,
-              description: resource.description,
-            })) ?? [],
+            response.results?.map((resource) => {
+              const returnObject = {
+                value: resource as PathValue<FormData, Name>,
+                description: resource.description,
+                label: resource.name,
+              };
+              if (
+                'summary_fields' in resource &&
+                typeof resource.summary_fields === 'object' &&
+                resource.summary_fields !== null &&
+                'credential_type' in resource.summary_fields &&
+                typeof resource.summary_fields.credential_type === 'object' &&
+                resource.summary_fields.credential_type !== null &&
+                'name' in resource.summary_fields.credential_type
+              ) {
+                returnObject.label = `${resource.name} | ${resource.summary_fields.credential_type.name as string}`;
+              }
```

Comment:
Instead of adding specific logic for credential resources in this shared component, we could allow label customization through props. We could pass in a custom function, such as `formatLabel`, and use that with `PageFormMultiSelectAwxResource`.

```
const returnObject = {
  value: resource as PathValue<FormData, Name>,
  description: resource.description,
  label: props.formatLabel ? props.formatLabel(resource) : resource.name,
};
```